R/oauth.R

```diff
@@ -48,10 +57,56 @@ auth_oauth_sign <- function(req, reauth = FALSE) {
       } else {
         token <- token_refresh(flow_params$client, token$refresh_token)
```

Comment:
Currently, I have a cached token with dually-expired values (after coming back from vacation):

1. `access_token` (expired after X hrs, <1 day)
2. `refresh_token` (expired after Y days, e.g., ~1 week)

Accordingly, `token_refresh()` code fails with the error:
```
Error in `token_refresh()`:
! OAuth failure [invalid_grant]
• The refresh token is invalid or expired.
```

Would it be reasonable to add a `tryCatch()` (or otherwise) such that if `token_refresh()` fails, per the cached `token$refresh`, then we fall back to `token <- exec(flow, !!!flow_params)`?

Reply:
Looks like #168 shared by @jennybc handles `N` refreshes with a _valid_ refresh token. 

I'm curious if you've thought about how _expired_ refresh tokens are handled?

Reply:
I have no thought about expired refresh tokens at all 😭 But I think it seem reasonable to try again here if `token_refresh()` fails with a `httr2_oauth` error.  Or maybe even `httr2_oauth_invalid_grant`?

Does the API you're using happen to document this somewhere I can read about it?

Reply:
From your other issues, maybe https://developer.okta.com/docs/guides/refresh-tokens/main/#refresh-token-rotation ?

Reply:
Yes, exactly! We are using refresh token rotation per those docs, with a set expiration of N days. So, we'd expect to `token_refresh()` to error on days ≥ N+1.

Reply:
I also noticed from the google docs that this also applies there, although google token expires after 6 months of no use, and httr2 deletes cached tokens after 30 days.

Reply:
nice! Thanks again for adding this @hadley, it's saved us a lot of code overhead and made the auth process ~trivial!